packages/grid/_modules_/grid/hooks/features/columns/useColumns.ts

```diff
@@ -226,5 +232,13 @@ export function useColumns(columns: Columns, apiRef: ApiRef): InternalColumns {
 
   useApiMethod(apiRef, colApi, 'ColApi');
 
+  const onResize = React.useCallback(
+    (size) => {
+      resetColumns(size.width);
+    },
+    [resetColumns],
+  );
+  useApiEventHandler(apiRef, RESIZE, onResize);
```

Comment:
From what I understand `gridState.containerSizes?.viewportSize.width` already depends on the resize event, so this whole block is duplicated logic.